Mathlib/MeasureTheory/Measure/Typeclasses.lean

```diff
@@ -1139,6 +1139,30 @@ instance [SigmaFinite (μ.restrict s)] : SigmaFinite (μ.restrict (s ∩ t)) :=
 instance [SigmaFinite (μ.restrict t)] : SigmaFinite (μ.restrict (s ∩ t)) :=
   sigmaFinite_of_le (μ.restrict t) (restrict_mono_ae (ae_of_all _ Set.inter_subset_right))
 
+lemma sigmaFinite_iUnion (μ : Measure α) {s : ℕ → Set α} (h_meas : MeasurableSet (⋃ n, s n))
```

Comment:
I guess you shouldn't need `h_meas` here: by `restrict_toMeasurable_of_sFinite`, each measure `μ.restrict (toMeasurable (s n))` is sigma-finite, so by the current version of the lemma then `μ.restrict (⋃ n, toMeasurable (s n))` is sigma-finite, and by monotonicity `μ.restrict (⋃ n, s n)` is also sigma-finite. Does it make sense?

Reply:
That seems to work only if `μ` is s-finite (in order to use `restrict_toMeasurable_of_sFinite`).
But I can definitely add a version of the lemma with that hypothesis instead.